lab_4_retrieval_w_clustering/main.py

```diff
@@ -305,6 +464,37 @@ def infer(self, query_vector: Vector, n_neighbours: int) -> list[tuple[float, in
         Returns:
             list[tuple[float, int]]: Distance to relevant document and document index.
         """
+        if not query_vector or not isinstance(query_vector, tuple):
+            raise ValueError("Check, please, query vector input in infer")
+        if not isinstance(n_neighbours, int) or n_neighbours < 0:
+            raise ValueError("Check, please, n_neighbours input in infer")
+
+        dict_for_clusters = {}
+        no_centroid = True
+        for cluster in self.__clusters:
+            centroid = cluster.get_centroid()
+            if not centroid:
+                no_centroid = False
+            distance = calculate_distance(query_vector, centroid)
+            if distance is None:
+                raise ValueError("Incorrect calculation, please check it")
+            dict_for_clusters[cluster] = distance
+
+        if no_centroid is False:
+            min_cluster = self.__clusters[0]
+        else:
+            min_cluster = min(dict_for_clusters, key=dict_for_clusters.get)
+
+        indices = min_cluster.get_indices()
+        vectors = self._db.get_vectors(indices)
+        list_for_vectors = []
+        for vector in vectors:
+            vector_distance = calculate_distance(query_vector, vector[-1])
+            if vector_distance is None:
+                raise ValueError("Incorrect calculation, please check it")
```

Comment:
I imagine I'm an ordinary stupid user and I run the program, and I get that I have the incorrect calculation. I would be shocked by this information